cmd/ctl/pkg/renew/renew.go

```diff
@@ -0,0 +1,212 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package renew
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	apiutil "github.com/jetstack/cert-manager/pkg/api/util"
+	cmapi "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha2"
+	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+	cmclient "github.com/jetstack/cert-manager/pkg/client/clientset/versioned"
+)
+
+var (
+	long = templates.LongDesc(i18n.T(`
+	Mark cert-manager Certificate resources for manual renewal.
+`))
+
+	example = templates.Examples(i18n.T(`
+		# Renew the Certificates named 'my-app' and 'vault' in the current context namespace.
+		ctl renew my-app vault
+
+		# Renew all Certificates in the 'kube-system' namespace.
+		ctl renew --namespace kube-system --all
+
+		# Renew all Certificates in all namespaces that have the label 'app=my-service'.
+		ctl renew --all-namespaces -l app=my-service`))
+)
+
+// Options is a struct to support renew command
+type Options struct {
+	// The Namespace that the Certificate to be renewed resided in
+	Namespace  string
+	CMClient   cmclient.Interface
+	RestConfig *restclient.Config
+
+	LabelSelector string
+	All           bool
+	AllNamespaces bool
+
+	genericclioptions.IOStreams
+}
+
+// NewOptions returns initialized Options
+func NewOptions(ioStreams genericclioptions.IOStreams) *Options {
+	return &Options{
+		IOStreams: ioStreams,
+	}
+}
+
+// NewCmdRenew returns a cobra command for renewing Certificates
+func NewCmdRenew(ioStreams genericclioptions.IOStreams, factory cmdutil.Factory) *cobra.Command {
+	o := NewOptions(ioStreams)
+
+	cmd := &cobra.Command{
+		Use:     "renew",
+		Short:   "Mark a Certificate for manual renewal",
+		Long:    "Mark a Certificate for manual renewal",
+		Example: example,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(factory))
+			cmdutil.CheckErr(o.Validate(args))
+			cmdutil.CheckErr(o.Run(args))
+		},
+	}
+
+	cmd.Flags().StringVarP(&o.LabelSelector, "selector", "l", o.LabelSelector, "Selector (label query) to filter on, supports '=', '==', and '!='.(e.g. -l key1=value1,key2=value2)")
+	cmd.Flags().BoolVarP(&o.AllNamespaces, "all-namespaces", "A", o.AllNamespaces, "If present, mark Certificates across namespaces for manual renewal. Namespace in current context is ignored even if specified with --namespace.")
+	cmd.Flags().BoolVar(&o.All, "all", o.All, "Renew all Certificates in the given Namespace, or all namespaces with --all-namespaces enabled.")
```

Comment:
What about `Namespace`?

Reply:
Covered by 

`	o.Namespace, _, err = f.ToRawKubeConfigLoader().Namespace()` 

and gets exposed through kubeconfig flags